src/osemgrep/cli_publish/Test_publish_subcommand.ml

```diff
@@ -0,0 +1,195 @@
+(* Brandon Wu
+ *
+ * Copyright (C) 2023 Semgrep, Inc.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License
+ * version 2.1 as published by the Free Software Foundation, with the
+ * special exception on linking described in file LICENSE.
+ *
+ * This library is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the file
+ * LICENSE for more details.
+ *)
+open Common
+open File.Operators
+open Testutil
+module Http_helpers = Http_helpers.Make (Lwt_platform)
+
+(*****************************************************************************)
+(* Prelude *)
+(*****************************************************************************)
+(* Port of test_publish.py to OCaml.
+ *
+ * Note that unlike most cli/tests/e2e/test_xxx.py tests, we can't reuse
+ * test_login.py to test osemgrep because of the use of mocking
```

Comment:
don't copy comments, or if you do at least update the comment, here s/test_login.py/test_publish.py